crates/matrix-sdk-indexeddb/src/state_store/mod.rs

```diff
@@ -1432,7 +1440,19 @@ impl_state_store!({
             .map(|item| QueuedEvent {
                 event: item.event,
                 transaction_id: item.transaction_id,
-                is_wedged: item.is_wedged,
+                error: match item.is_wedged {
+                    Some(legacy_wedged) => {
+                        if legacy_wedged {
+                            // migrate a generic error
+                            Some(QueueWedgeError::GenericApiError {
+                                msg: "local echo failed to send in a previous session".into(),
+                            })
+                        } else {
+                            None
+                        }
+                    }
+                    None => item.error,
```

Comment:
This doesn't correctly handle the case where legacy is set to false, but there's still a new-form error. I think you want something like:

```rust
Some(true) => // migrate,
_ => item.error
```